doc/source/programs/gdal_contour.rst

```diff
@@ -146,12 +155,111 @@ Examples
+            { "type": "Feature", "properties": { "ID": 2, "min": 20.0, "max": 30.0 }, "geometry": { "type": "MultiPolygon", "coordinates": [ [ [ [ 0.954545454545455, 0.0 ], [ 0.954545454545455, 0.5 ], [ 1.5, 0.785714285714286 ], [ 2.0, 0.785714285714286 ], [ 2.0, 1.0 ], [ 2.0, 1.261904761904762 ], [ 1.5, 1.261904761904762 ], [ 0.5, 0.738095238095238 ], [ 0.0, 0.738095238095238 ], [ 0.0, 0.5 ], [ 0.0, 0.0 ], [ 0.5, 0.0 ], [ 0.954545454545455, 0.0 ] ] ] ] } },
+            { "type": "Feature", "properties": { "ID": 3, "min": 30.0, "max": 36.0 }, "geometry": { "type": "MultiPolygon", "coordinates": [ [ [ [ 1.499999909090926, 0.0 ], [ 1.0, 0.0 ], [ 0.954545454545455, 0.0 ], [ 0.954545454545455, 0.5 ], [ 1.5, 0.785714285714286 ], [ 2.0, 0.785714285714286 ], [ 2.0, 0.500000047619043 ], [ 1.5, 0.500000047619043 ], [ 1.499999909090926, 0.5 ], [ 1.499999909090926, 0.0 ] ] ] ] } }
+            ]
+        }
+
+.. example::
+
+    :title: Creating contours from a DEM with fixed levels
+
+    .. code-block:: bash
+
+        $ cat test.asc
+        ncols        2
+        nrows        2
+        xllcorner    0
+        yllcorner    0
+        cellsize     1
+        4 15
+        25 36
+
+        $ gdal_contour test.asc -f GeoJSON /vsistdout/ -fl 10 20 -p -amin min -amax max
+
+    This would create a single polygonal contour between 10 and 20 meters from the DEM data in :file:`test.asc`
+    and produce a GeoJSON output with the contour min and max elevations in the ``min`` and ``max`` attributes.
+.
+
+    If the minimum and maximum values from the raster are desired, the special values `MIN`` and `MAX``
+    (case insensitive) can be used:
+
+    .. code-block:: bash
+
+        $ cat test.asc
+        ncols        2
+        nrows        2
+        xllcorner    0
+        yllcorner    0
+        cellsize     1
+        4 15
+        25 36
+
+        $ gdal_contour test.asc -f GeoJSON /vsistdout/ -fl MIN 10 20 MAX -p -amin min -amax max
+
+    This would create three polygonal contours from the DEM data in :file:`test.asc` and produce a GeoJSON output
+    with the contour min and max elevations in the ``min`` and ``max`` attributes, the values of these fields will
+    be: (4.0, 10.0), (10, 20.0) and (20, 36.0).
+
+.. example::
+
+    :title: Creating contours from a DEM specifying an interval and fixed levels at the same time
+
+    .. code-block:: bash
+
+        $ cat test.asc
+        ncols        2
+        nrows        2
+        xllcorner    0
+        yllcorner    0
+        cellsize     1
+        4 15
+        25 36
+
+        $ gdal_contour test.asc -f GeoJSON /vsistdout/ -i 10 -fl 15 -p -amin min -amax max
+
+    This would create 10-meter polygonal contours from the DEM data in :file:`test.asc` and produce a GeoJSON output
+    with the contour min and max elevations in the ``min`` and ``max`` attributes, the values of these fields will
+    be: (4.0, 10.0), (10, 15.0), (15, 20.0)  and (20, 36.0).
 
```

Comment:
Shoud it be (10, 15.0), (15, 20.0)  and (20, 30.0)?

I think I understand what happens, but not how to describe it so that a newcomer understands. Perhaps something like
"Creates contours at regular 10 meter intervals and adds extra contour for a fixed 15 m level. Finally turns areas between the contours into polygons."

Reply:
For those values, shouldn't user define `-i 10 -fl MIN 15 MAX -p`? Otherwise it would not match with the title...

Reply:
Maybe a bit different approach would have been easier to explain. First improve documentation of -p

> "Generate contour polygons rather than contour lines. The min and max values of the raster will be used as the lower and upper limit for the data range of the first and last polygon, respectively.

Then introduce `exlude_min` and `exclude_max` for preventing the default behavior so that it would have similar effect on all `-i` and `-fl` combinations, when used together with `-p`, and no effect without `-p`.

Reply:
> For those values, shouldn't user define `-i 10 -fl MIN 15 MAX -p`? Otherwise it would not match with the title. Or do regular intervals start/stop at min/max automatically? It does make sense, most users want to get the polygon between the sea and 10 m level automatically, but it is not clear from the documentation.

yes, regular intervals get min/max automatically. I didn't change that.